python/ray/serve/api.py

```diff
@@ -357,6 +360,27 @@ class MyDeployment:
     if isinstance(logging_config, LoggingConfig):
         logging_config = logging_config.dict()
 
+    if num_replicas == "auto":
+        num_replicas = None
+        if max_concurrent_queries is DEFAULT.VALUE:
+            max_concurrent_queries = NEW_DEFAULT_MAX_CONCURRENT_QUERIES
+
+        if autoscaling_config in [DEFAULT.VALUE, None]:
```

Comment:
OOC why do we need to handle both?

and why _don't_ we need to for max_concurrent_queries?